src/qinfer/resamplers.py

```diff
@@ -61,7 +62,7 @@
 
 class Resampler(with_metaclass(ABCMeta, object)):
     @abstractmethod
-    def __call__(self,  model, particle_weights, particle_locations,
+    def __call__(self,  model, particle_dist,
```

Comment:
I also take back what I said about the class method, I think it makes the most sense as is.